### PR TITLE
Remove unnecesary clone in `Form::class`, add `fieldset::class` widget.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Widget/Form.php [hunk 1]
```diff
@@ -197,6 +211,38 @@ public function enctype(string $value): self
...
+    /**
+     * The <fieldset> HTML element is used to group several controls as well as labels (<label>) within a web form.
+     *
+     * @param bool $value whether the fieldset is enabled or disabled.
+     *
+     * @return static
+     *
+     * @link https://html.spec.whatwg.org/multipage/form-elements.html#the-fieldset-element
+     */
+    public function fieldset(bool $value): self
```
Comment (samdark) [inline src/Widget/Form.php:223]: I don't think there's a point adding fieldset as a boolean flag that just wraps all the content. See https://jsfiddle.net/6fz3nvLr/ for typical usage. It is multiple fieldsets per form.

File: src/Widget/Form.php [hunk 2]
```diff
@@ -211,6 +257,38 @@ public function id(string $value): self
...
+    /**
+     * The <legend> HTML element represents a caption for the content of its parent <fieldset>.
+     *
+     * @param string|null $value whether the legend is enabled or disabled.
+     *
+     * @return static
+     *
+     * @link https://html.spec.whatwg.org/multipage/form-elements.html#the-legend-element
+     */
+    public function legend(?string $value): self
```
Comment (samdark) [inline src/Widget/Form.php:269]: Same as for fieldset.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Widget/Form.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Widget/Form.php:223]: yes, if it makes sense.

File: src/Widget/Form.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Widget/Form.php:269]: yes, if it makes sense.

File: src/Widget/Form.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Widget/Form.php]: fixed.

File: src/Widget/Form.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Widget/Form.php]: fixed.

Comment (terabytesoftw) [reply src/Widget/Form.php]: fixed.

**Comment by terabytesoftw** [comment]
Add tests in FieldsetTest:

image

File: src/Widget/Form.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Widget/Form.php]: done.

**Review by samdark — APPROVED** [review]

**Pull request merged**